### ISSUE TITLE
Feature Request: Utilize LibDBIcon's API for AoC Menu

### ISSUE THREAD

**Issue by StormFX** [WeakAuras/WeakAuras2#4453]
**Is your feature request related to a problem? Please describe.**

Currently, the WA menu button in the add-on compartment can't be disabled. This can potentially cause taint. Additionally, the tooltip covers the menu.

**Describe the solution you'd like**

Allow the middle-mouse to cycle between no icon, the minimap icon _or_ the add-on compartment, as having one of the latter two enabled makes the other redundant.

An alternative (and better) solution would be to use LibDBIcon-1.0's updated API to handle registering and unregistering the AoC button and tooltip. This allows a simpler method of handling the placement options while also solving the issue with the tooltip.

**Describe alternatives you've considered**

N/A

**Additional context**

Current:
![image](https://github.com/WeakAuras/WeakAuras2/assets/1234928/88747dd7-ce26-43ed-83c0-374fb0f66f66)

Using LibDBIcon (Masque):
![image](https://github.com/WeakAuras/WeakAuras2/assets/1234928/da36546f-8f11-4e95-98bf-ff5e700f47c5)

**Comment by Stanzilla** [comment]
Yeah I think a tri-state dropdown would be the best here. One day I will have to code up the default settings panel...

**Comment by StormFX** [comment]
You mean something besides just the aura management panel with some buttons on the frame?  :p Could add a "Settings" tab and clean up the main panel a bit.

**Comment by Stanzilla** [comment]
> You mean something besides just the aura management panel with some buttons on the frame? :p Could add a "Settings" tab and clean up the main panel a bit.

basically this

**Comment by StormFX** [comment]
I'm looking to something similar with Masque, just going to take a _lot_ of work. I'm wanting to do something similar to the work order frame, but will have to look into keeping it Classic-friendly. In any case, I'm sure it'll be awesome! Thanks!
